### PR TITLE
Add table view only to chart if really shown

### PR DESCRIPTION
**Pull request by dedeibel**
- Mitigates problem of it taking control over the charts size

I talked about it with akrimm, please ask if you want more info.

### CODE REVIEW

**Review by wirew0rm — DISMISSED** [review]
:+1:
Looks good, thanks.
Will be merged as soon as travis runs though.

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

File: chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        tableViewer = new TableViewer();
+        chart.setPrefWidth(400);
+        chart.setPrefHeight(300);
+        chart.getDatasets().add(new CosineFunction("Cosine", 50));
+        stage.setScene(scene);
+        stage.show();
+    }
+
+    @Test
+    public void testThatTableIsOnlyAddedToSceneWhenActive() throws TimeoutException {
```
Comment (RalphSteinhagen) [inline chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java:54]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/test/java/de/gsi/chart/plugins/TableViewerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6378322&bid=20691195)

**Review by wirew0rm — APPROVED** [review]
Looks good, thanks also for the nice tests and the patience with the CI :)

**Pull request merged**